### PR TITLE
fix: render condition for review button in search result page

### PR DESCRIPTION
**Pull request by tahmidrahman-dsi**
image

image

### CODE REVIEW

File: packages/client/src/views/SearchResult/SearchResult.tsx [hunk 1]
```diff
@@ -313,6 +315,7 @@ class SearchResultView extends React.Component<
               !declarationIsRegistered &&
               !declarationIsCertified &&
               !declarationIsArchived &&
+              !declarationIsCorrectionRequested &&
```
Comment (rikukissa) [inline packages/client/src/views/SearchResult/SearchResult.tsx:318]: Could you clean up this conditional here a little bit please? Just by reading this I have a very hard time understanding when exactly does this conditional apply. I see you added a check that the record is not in "correction requested" state but then on line 328 there's code to handle that

Comment (tahmidrahman-dsi) [reply packages/client/src/views/SearchResult/SearchResult.tsx]: @rikukissa could you please have a look?  [6dc8606](https://github.com/opencrvs/opencrvs-core/pull/7226/commits/6dc86068854c56712903346549043ef54cc3e386)

**Comment by Zangetsu101** [comment]
@tahmidrahman-dsi The issue mentioned here had something to do with the download functionality: https://github.com/opencrvs/opencrvs-core/issues/6247#issuecomment-2182473081 Has that been fixed here?

**Comment by tahmidrahman-dsi** [comment]
> @tahmidrahman-dsi The issue mentioned here had something to do with the download functionality: https://github.com/opencrvs/opencrvs-core/issues/6247#issuecomment-2182473081 Has that been fixed here?

Yes that was fixed, but there was also another comment there https://github.com/opencrvs/opencrvs-core/issues/6247#issuecomment-2182473081 , this fix corresponds to that comment

File: packages/client/src/views/SearchResult/SearchResult.tsx [hunk 2]
```diff
@@ -281,6 +283,18 @@ class SearchResultView extends React.Component<
...
           params.get('searchType')
         ]
+        const isDeclarationReviewableByRegistrar =
+          declarationIsValidated ||
+          declarationIsCorrectionRequested ||
+          isDeclared ||
+          declarationIsInProgress
+
+        const isDeclarationReviewableByRegAgent =
+          isDeclared || declarationIsInProgress || declarationIsRejected
```
Comment (Zangetsu101) [inline packages/client/src/views/SearchResult/SearchResult.tsx:293]: not too sure about the rejected one being reviewable by RA :sweat_smile:

Comment (tahmidrahman-dsi) [reply packages/client/src/views/SearchResult/SearchResult.tsx:294]: Thanks @Zangetsu101 , as I was also not completely sure, I verified with Syeda that a RA can review rejected declarations. In addition to that I also I added to registrar review condition here in this commit https://github.com/opencrvs/opencrvs-core/pull/7226/commits/75484b87b550ffefff5a84b461c467e02c028ad1

Comment (rikukissa) [reply packages/client/src/views/SearchResult/SearchResult.tsx:294]: This is so much cleaner now @tahmidrahman-dsi ! Good work.

**Review by rikukissa — APPROVED** [review]

**Comment by rikukissa** [comment]
@tahmidrahman-dsi @Zangetsu101 this ready to go?

**Comment by Zangetsu101** [comment]
> @tahmidrahman-dsi @Zangetsu101 this ready to go?

Should be

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**